internal/pipe/archive/archive.go

```diff
@@ -117,6 +117,9 @@ func create(ctx *context.Context, archive config.Archive, binaries []*artifact.A
 	}
 	archivePath := filepath.Join(ctx.Config.Dist, folder+"."+format)
 	lock.Lock()
+	if err := os.MkdirAll(filepath.Dir(archivePath), 0755|os.ModeDir); err != nil {
+		return err
```

Comment:
hey, I think you missed `lock.Unlock()` here

Reply:
thanks!